next/frontend/utils/formRoutes.ts

```diff
@@ -0,0 +1,177 @@
+import { formsApi } from '@clients/forms'
+import { GetFormResponseDtoStateEnum } from '@clients/openapi-forms'
+import { isAxiosError } from 'axios'
+import { GetServerSidePropsContext, GetServerSidePropsResult } from 'next/types'
+import { serverSideTranslations } from 'next-i18next/serverSideTranslations'
+
+import { FormPageWrapperProps } from '../../components/forms/FormPageWrapper'
+import { getSSRCurrentAuth } from '../../components/logic/ServerSideAuthProvider'
+import { ROUTES } from '../api/constants'
+
+export const existingFormServerSideProps = async (
+  ctx: GetServerSidePropsContext,
+  slug: string,
+  formId: string,
+): Promise<GetServerSidePropsResult<FormPageWrapperProps>> => {
+  const ssrCurrentAuthProps = await getSSRCurrentAuth(ctx.req)
+
+  try {
+    const [form, files] = await Promise.all([
+      formsApi
+        .nasesControllerGetForm(formId, {
+          accessToken: 'onlyAuthenticated',
+          accessTokenSsrReq: ctx.req,
+        })
+        .then((res) => res.data),
+      formsApi
+        .filesControllerGetFilesStatusByForm(formId, {
+          accessToken: 'onlyAuthenticated',
+          accessTokenSsrReq: ctx.req,
+        })
+        .then((res) => res.data),
+    ])
+
+    if (
+      !form ||
+      /* If there wouldn't be this check it would be possible to open the page with any slug in the URL. */
```

Comment:
wouldn't the !form (and more importantly, the request throwing with 404) catch such cases ? or does backedn return non-empty `form` in these cases ?